src/server.rs

```diff
@@ -32,24 +32,41 @@ use RpcContext;
 
 const DEFAULT_REQUEST_SLOTS_PER_CQ: usize = 1024;
 
-pub type CallBack = Box<Fn(RpcContext, &[u8])>;
-
 /// An RPC call holder.
-pub struct Handler {
+#[derive(Clone)]
+pub struct Handler<F> {
     method_type: MethodType,
-    cb: CallBack,
+    cb: F,
 }
 
-impl Handler {
-    pub fn new(method_type: MethodType, cb: CallBack) -> Handler {
+impl<F> Handler<F> {
+    pub fn new(method_type: MethodType, cb: F) -> Handler<F> {
         Handler { method_type, cb }
     }
+}
+
+pub trait CloneableHandler: Send {
```

Comment:
I think you could add the `Clone` trait bound here?

Reply:
No, `CloneableHandler` is going to be used as trait object, and `Clone` is not object safe.